app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -493,18 +500,10 @@ def vm_hardware_guest_devices(persister_hardware, vm, addresses, host)
       mac = nic.mac && nic.mac.address ? nic.mac.address : nil
       network = mac && networks.present? ? networks[mac] : nil
 
-      profile_id = nic.dig(:vnic_profile, :id)
-      profiles = collector.collect_vnic_profiles
-      vnic_profile = profiles.detect { |p| p.id == profile_id } if profile_id && profiles
-      network_id = vnic_profile.dig(:network, :id) if vnic_profile
-      lan = if network_id && host
-              switch = persister.host_virtual_switches.lazy_find(
-                :host    => host,
-                :uid_ems => network_id
-              )
-
-              persister.lans.lazy_find(:switch => switch, :uid_ems => network_id)
-            end
+      vnic_profile_id = nic.dig(:vnic_profile, :id)
+      network_uid = collector.collect_vnic_profiles.detect { |vp| vp.id == vnic_profile_id }&.network&.id
```

Comment:
Is this `collect_vnic_profiles` cached?  It doesn't look like it and this could lead to a huge number of API calls to the provider (one per vm nic).

How about a method like `collector#vnic_profiles` which does `@vnic_profiles ||= collect_vnic_profiles`

Reply:
You are right. I changed it when dealing with a bug, will make it great again.

Reply:
lol thanks :+1:

Reply:
I just added it to the collect_vnic_profiles, is that ok?

Reply:
:+1: